### PR TITLE
ENH: Implementation of Bland-Altman plot

### PR DESCRIPTION
**Pull request by josesho**
Implementation of a [Bland-Altman plot](https://en.wikipedia.org/wiki/Bland%E2%80%93Altman_plot) that is heavily used in method comparison for medical statistics.

This type of plot is different from the [Tukey HSD plot](http://www.statsmodels.org/dev/generated/statsmodels.sandbox.stats.multicomp.TukeyHSDResults.plot_simultaneous.html) that is currently under dev here, and thus could be added on its own.

### CODE REVIEW

**Comment by josef-pkt** [comment]
appveyor problems are unrelated, has one conda problems, one has #4108 

@josesho  Thanks for the contribution

The main issue I have with this is the name and location. In general it is better to have informative names for functions or objects instead of using proper names. In this case `bland_altman` might not be informative enough except for users in the fields where it is used under this name.

Another name for this or a version of this according to Wikipedia is Tukey's mean difference plot.
So one possibility would be to rename the module to `agreement.py` and the function for example to `mean_diff_plot`.
Do you have other suggestions?

Otherwise it looks fine in a quick read-through. There are a few details that need changes, e.g. imports on top of module.

**Comment by josesho** [comment]
I'm ok with the name changes for the module and the function. Will clean up the module itself. 

I had some issues building the docs locally, but let me incorporate the name changes and we can fix the docs build.

Thanks!
Joses

**Comment by josef-pkt** [comment]
renaming is still incomplete, travis fails everywhere

```
statsmodels/graphics/api.py:12: in <module>
    from .bland_altman import bland_altman_plot
E   ImportError: No module named 'statsmodels.graphics.bland_altman'
```

File: statsmodels/graphics/agreement.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+# Author: Joses W. Ho
+
+import numpy as np
+import matplotlib.pyplot as plt
```
Comment (josef-pkt) [inline statsmodels/graphics/agreement.py]: still unprotected matplotlib import
matplotlib is an optional dependency

File: statsmodels/graphics/tests/test_mean_diff_plot.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+from statsmodels.compat.testing import skipif
+
+import numpy as np
+
+from statsmodels.agreement import mean_diff_plot
```
Comment (josef-pkt) [inline statsmodels/graphics/tests/test_mean_diff_plot.py]: travis fails, Import needs full paths
`from statsmodels.graphics.agreement import ...`

File: statsmodels/graphics/agreement.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    if sd_limit < 0:
+        raise ValueError('sd_limit ({}) is less than 0.'.format(sd_limit))
+
+    means = np.mean([m1, m2], axis=0)
+    diffs = m1 - m2
+    mean_diff = np.mean(diffs)
+    std_diff = np.std(diffs, axis=0)
+
+    if ax is None:
+        ax = plt.gca()
```
Comment (josef-pkt) [inline statsmodels/graphics/agreement.py]: we have a helper function for creating a figure with axis. 
we don't use/attach to `gca`, users can use `ax` if they want that.

File: statsmodels/graphics/tests/test_mean_diff_plot.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+try:
+    import matplotlib.pyplot as plt
+    have_matplotlib = True
+except:
+    have_matplotlib = False
+
+
+@skipif(not have_matplotlib, reason='matplotlib not available')
+def test_mean_diff_plot():
```
Comment (josef-pkt) [inline statsmodels/graphics/tests/test_mean_diff_plot.py]: needs also unit test with pandas Series or DataFrame. 
There is no `asarray` to convert m1, m2, but most likely the plot works with Series

**Comment by josef-pkt** [comment]
Looks close to finished. I would like to merge it soon, and could also do some final changes if needed.

Needs rebase and some squashing, I think complete squashing into a single commit would be fine here.
I can do the interactive rebase if you are not familiar with those.

File: statsmodels/graphics/agreement.py (hunk 3 above)
Comment (josesho) [reply statsmodels/graphics/agreement.py]: OK, let me look at the other plotting functions and modify this.

**Comment by josesho** [comment]
> Needs rebase and some squashing, I think complete squashing into a single commit would be fine here. I can do the interactive rebase if you are not familiar with those.

That would be good, I am clueless about the interactive rebasing!

Thanks,
Joses

**Comment by josef-pkt** [comment]
test failure
test_mean_diff_plot is missing pandas import

@josesho Thanks, I can finish this up for merge and do the rebase within the next few days. I will give it another brief look, but I think it is essentially ready.

**Pull request closed without merging**